src/link/C.zig

```diff
@@ -13,15 +13,29 @@ const C = @This();
 
 pub const base_tag: File.Tag = .c;
 
+pub const HeaderInfo = struct {
```

Comment:
The header specific stuff uses this instead of a `link.C`. In the normal codegen case, we pass in `link.C.header_info` and when we're generating C headers, we create our own `HeaderInfo`.